modules/minimint-mint/src/lib.rs

```diff
@@ -279,23 +284,26 @@ impl FederationModule for Mint {
                         }));
 
                         batch_tx.append_insert(OutputOutcomeKey(issuance_id), blind_signature);
-                        batch_tx.commit();
-                        Some(batch)
                     }
-                    Err(CombineError::TooFewShares(got, need)) => {
-                        trace!(
-                            %issuance_id,
-                            got,
-                            need,
-                            "Failed to combine signature shares: not enough shares",
-                        );
-                        None
+                    Err(CombineError::TooFewShares(got, _)) => {
+                        for non_contributing in consensus_peers.sub(&HashSet::from_iter(got)) {
+                            error!(
+                                "Dropping {:?} for not contributing shares",
+                                non_contributing
+                            );
+                            batch_tx.append_insert(DropPeerKey(non_contributing), ());
```

Comment:
Using the DB for communication between modules and consensus isn't ideal imo. My rule of thumb is "mdules should still work if they had separate databases".

Reply:
True, I don't know why I didn't think of just returning it from `end_consensus_epoch`